docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,268 @@
-harder to construct. To improve the situation, Flux provides a different configuration variant: `patchUpdated`.
-`patchUpdated` configurations store the modifications from Flux into a
-[YAML merge patch file](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md)
-and implicitly apply them to the resources printed by the `generators`.
+### Execution context of commands
 
-Here is an example, allowing to deploy a [Helm chart without a Tiller installation](https://medium.com/@jeroen.rosenberg/using-helm-charts-without-tiller-f1588bc1f0c4)
+`generators` and `updaters` are run in a POSIX shell inside the fluxd
+container. This means that the executables mentioned in commands must
+be available in the [Flux container
+image](https://github.com/fluxcd/flux/blob/master/docker/Dockerfile.flux).
 
-```yaml
-version: 1
-patchUpdated:
-  generators:
-    - command: helm template ../charts/mychart -f overrides.yaml
-  patchFile: flux-patch.yaml
-```
+Flux currently includes `kustomize` and basic Unix shell tools. If the
+tools in the Flux image are not sufficient for your use case, you can
+include new tools in your own Flux-based image or, if the tools are
+popular enough, Flux maintainers can add them to the Flux image
+(please create an issue). In the future it may be possibly to specify
+an container image for each command.
 
-This configuration will cause:
-1. modifications made by Flux (e.g. due to annotation or image updates) to the resources generated by `helm template ../charts/mychart -f overrides.yaml` to be stored into patch file `flux-patch.yaml`. `flux-patch.yaml` will be managed automatically by Flux (it will be added and updated on demand).
-2. the patch file `flux-patch.yaml` to be always applied to the output of `helm template ../charts/mychart -f overrides.yaml`, in order to obtain the final manifests.
+The working directory (also known as CWD) of the `command`s executed
+from a `.flux.yaml` file will be set to the target path, i.e., the
+`--git-path` entry.
 
-The patch file path should be relative to Flux target which matched the configuration file.
+For example, when using flux with `--git-path=staging` on a git
+repository with this structure:
 
-Note that the patch file will need to be kept consistent with any changes made in the generated manifests. In particular,
-the patch file will be sensitive to changes in workload names, workload namespaces or workload kinds.
+```sh
+├── .flux.yaml
+├── staging/
+├──── [...]
+├── production/
+└──── [...]
+```
 
-Lastly, here is another example using Kustomize which is much simpler than the `commandUpdated`-based example presented
-earlier.
+The commands in `.flux.yaml` will be executed with their working
+directory set to `staging`.
 
-```yaml
-version: 1
-patchUpdated:
-  generators:
-    - command: kustomize build .
-  patchFile: flux-patch.yaml
-```
+In addition, `updaters` are given arguments via environment variables:
```

Comment:
Nitpick:

```suggestion
In addition, the `updaters` of the `commandUpdated` configuration are given arguments via environment variables:
```

Reply:
I would include an example of `kustomize edit set image`

Reply:
I did not attempt to give a working example of `commandUpdated` updaters, because the thought of trying to formulate one terrified me. `kustomize edit set imagetag` sets the image for _all things that use it_, last time I looked.